src/core/positioning/egenioussreceiver.cpp

```diff
@@ -9,37 +9,34 @@ EgenioussReceiver::EgenioussReceiver( QObject *parent )
 {
   connect( mTcpSocket, &QTcpSocket::readyRead, this, &EgenioussReceiver::onReadyRead );
   connect( mTcpSocket, &QTcpSocket::errorOccurred, this, &EgenioussReceiver::handleError );
-  connect( mTcpSocket, &QTcpSocket::connected, this, &EgenioussReceiver::connected );
-  connect( mTcpSocket, &QTcpSocket::disconnected, this, &EgenioussReceiver::disconnected );
+  connect( mTcpSocket, &QTcpSocket::stateChanged, this, &EgenioussReceiver::setSocketState );
   setValid( true );
 }
 
-void EgenioussReceiver::handleConnectDevice()
+EgenioussReceiver::~EgenioussReceiver()
 {
-  mTcpSocket->connectToHost( mAddress, mPort, QTcpSocket::ReadWrite );
+  disconnect( mTcpSocket, &QTcpSocket::stateChanged, this, &EgenioussReceiver::setSocketState );
```

Comment:
Here you should explicitly disconnect the socket if connected. I think explicitly disconnecting the signal is not necessary